spec/system/planning_applications/assessment_against_legislation_spec.rb

```diff
@@ -193,16 +188,18 @@
           "Comment added on 01 Sep 2022 by Alice Smith",
           with: ""
         )
-      end
 
-      within(row_with_content("D.1a")) do
         fill_in(
           "Comment added on 01 Sep 2022 by Alice Smith",
           with: "Updated comment"
         )
       end
 
       click_button("Save and come back later")
+
+    rescue Selenium::WebDriver::Error::UnexpectedAlertOpenError
```

Comment:
I wonder if there's a different way of structuring this. If I'm understanding it correctly, there are different `expect` statements depending on whether the exception is raised or not?

Or is it that the exception will _always_ fire and so L182–188 _and_ L206–217 will have effect?

I wonder if we could limit the `rescue` block to only around the line that's expected to raise the exception, and/or use an `ensure` block so that it's clear what will happen if the exception is/isn't raised.

Reply:
My thinking was just to rescue this error and do nothing as it's not relevant to the test itself.

Basically navigating away from that page i.e. doing `visit planning_application_assessment_tasks_path(planning_application)` or `click_link("Part 1, Class D")` intermittently causes that error to raise when it shouldn't.  

I've spent a bit too much time on this though so if you could pull it locally and give it a whirl to your suggestion? I've tried wrapping it in a begin block but i think more than one action actually triggers this weird error so it's hard to isolate. I'm not sure it's worth the pain/effort to go deeper in a rabbit hole though.

Reply:
A good metric was to see if this would all be green `repeat 10 { bundle exec rspec spec/system/planning_applications/assessment_against_legislation_spec.rb:142 }`

Reply:
> My thinking was just to rescue this error and do nothing as it's not relevant to the test itself.

That's what confused me — it seems like actually the error is being rescued and then part of the test is done in the rescue block.

But if the exception can be thrown at multiple points then that might complicate matters.

I'd be happy to take a look at it myself this morning.

Reply:
I had thought perhaps `accept_alert { … }` might get us somewhere but even when the code is wrapped in that block the exception is thrown.